scripts/prepdocs.py

```diff
@@ -271,6 +287,36 @@ def create_search_index():
     else:
         if args.verbose: print(f"Search index {args.index} already exists")
 
+def update_embeddings_in_batch(sections):
+    batch_queue = []
+    copy_s = []
+    batch_response = {}
+    token_count = 0
+    for s in sections:
+        token_count += calculate_tokens_emb_aoai(s["content"])
+        if token_count <= MAX_EMB_TOKEN_LIMIT and len(batch_queue) < MAX_BATCH_SIZE:
+            batch_queue.append(s)
+            copy_s.append(s)
+        else:
+            emb_responses = compute_embedding_in_batch([item["content"] for item in batch_queue])
+            if args.verbose: print(f"Batch Completed. Batch size  {len(batch_queue)} Token count {token_count}")
+            for emb, item in zip(emb_responses, batch_queue):
+                batch_response[item["id"]] = emb
+            batch_queue = []
+            batch_queue.append(s)
+            token_count = calculate_tokens_emb_aoai(s["content"])
```

Comment:
Is this `token_count` calculation used? I'm trying to understand line 307 vs line 296.